tests/test_compose_curated_destination.py

```
@@ -0,0 +1,538 @@
+            calibration.write_text('{"records":[]}\n', encoding="utf-8")
+            moved_parent = root / "original-calibration-parent"
+            replacement_parent = root / "replacement-calibration-parent"
+            replacement_parent.mkdir()
+            (replacement_parent / calibration.name).write_text(
+                '{"records":[{"scope":"ffpc-r99-a","usd_conversion_factor":2}]}\n',
+                encoding="utf-8",
+            )
+            real_open = os.open
+            swapped = False
+
+            def swap_parent_before_open(path, flags, mode=0o777, *, dir_fd=None):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `swap_parent_before_open`

(ANN202)